### PR TITLE
Add spelling workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
- [x] The contributed code is licensed under GPL v2.0
- [x] Contributor Licence Agreement (CLA) is signed
- [ ] util/update-modules has been ran
- [ ] Documentation updated
- [ ] Test suite update

From https://github.com/percona/percona-toolkit/pull/660#issuecomment-1688198052

### CODE REVIEW

File: .github/workflows/toolkit.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ on:
   pull_request:
     branches: [ "3.x" ]
 
+concurrency:
```
Comment (szepeviktor) [inline .github/workflows/toolkit.yml:9]: I am not sure no permission `permissions: {}` will do here.

File: .typos.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+[files]
+extend-exclude = [
+    ".git/",
+    # "bin/",
```
Comment (szepeviktor) [inline .typos.toml:4]: @svetasmirnova bin contains a lot of typos fixed in lib.
Could it be that few of them were not updated?

File: src/go/pt-mongodb-query-digest/README.md [hunk 3]
```diff
@@ -1,49 +1,49 @@
...
-|-f|--output-format|report output format. Valid values are text, json. Default: text|
-|-l|--log-level|Log level:, panic, fatal, error, warn, info, debug error|
-|-n|--limit|show the first n queries|
-|-o|--order-by|comma separated list of order by fields (max values): `count`, `ratio`, `query-time`, `docs-scanned`, `docs-returned`.<br> A `-` in front of the field name denotes reverse order.<br> Example:`--order-by="count,-ratio"`).|
-|-p|--password[=password]|Password (optional). If it is not specified it will be asked|
-|-s|--skip-collections|Comma separated list of collections to skip. Default: `system.profile`. It is possible to use an empty list by setting `--skip-collections=""`|
-|-u|--user|Username|
-|-v|--version|Show version & exit|
-
+#pt-mongodb-query-digest
```
Comment (szepeviktor) [inline src/go/pt-mongodb-query-digest/README.md]: This README had CRLF line ends.

File: .typos.toml (hunk 2 above)
Comment (svetasmirnova) [reply .typos.toml:4]: I need to check. But all modules should not be touched in bin. However, some tool's code exists in the bin version only (it starts from "package TOOLNAME;" line), so having line ending checks in bin would be nice.

File: src/go/pt-mongodb-query-digest/README.md
Comment (svetasmirnova) [inline src/go/pt-mongodb-query-digest/README.md:1]: Please remove this file. It will be updated as part of PR 660

Comment (svetasmirnova) [reply src/go/pt-mongodb-query-digest/README.md:1]: Ah, I see - you removed 660. Resolving this comment.

Comment (szepeviktor) [reply src/go/pt-mongodb-query-digest/README.md:1]: Okay!
PR 660 is superseded by #662

File: .typos.toml (hunk 2 above)
Comment (szepeviktor) [reply .typos.toml:4]: > having line ending checks in bin would be nice.

@svetasmirnova I am starting with EditorConfig.
Line ends are fixed in #664
There are hundreds of files with trailing whitespaces.

Would you welcome a PR with those removed?
The single command would be [`eclint`](https://github.com/greut/eclint-action) with `-fix`.
```
eclint -exclude="{util/NaturalDocs/**,bin/**,t/**,util/pod2rst-fixed.packed}" -fix
```

538 files changed, 4532 insertions(+), 4532 deletions(-)

Comment (svetasmirnova) [reply .typos.toml:4]: Yes, please, but check if the tests are passed with Percona Server 5.7 and 8.0. @alina-derkach-oaza - do we use trailing whitespaces in docs for something?

Comment (alina-derkach-oaza) [reply .typos.toml:4]: @svetasmirnova, no, we usually delete trailing whitespaces in docs.

**Review by EvgeniyPatlan — APPROVED** [review]

File: .typos.toml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+"NDOnLoad" = "NDOnLoad"
+"NDOnResize" = "NDOnResize"
+"NULLable" = "NULLable"
+"O_WRONLY" = "O_WRONLY"
+"START_ND_SUMMARY" = "START_ND_SUMMARY"
+"START_ND_TOOLTIPS" = "START_ND_TOOLTIPS"
+"TOI" = "TOI"
+"UNIONed" = "UNIONed"
+"UNIONs" = "UNIONs"
+"WatchFileFoChanges" = "WatchFileFoChanges"
```
Comment (szepeviktor) [inline .typos.toml]: Isn't "WatchFileFoChanges" really a typo?
https://github.com/search?q=repo%3Apercona%2Fpercona-toolkit%20WatchFileFoChanges&type=code
There are many "WatchFileForChanges". (with an "r")

Comment (svetasmirnova) [reply .typos.toml]: I believe this is a typo, but again, this is not our code.

Comment (svetasmirnova) [reply .typos.toml]: Let's fix it anyway.

Comment (szepeviktor) [reply .typos.toml]: Done 🍏

**Review by svetasmirnova — APPROVED** [review]

**Pull request merged**